xrootd/client/session.go

```diff
@@ -103,23 +140,41 @@ func newSession(ctx context.Context, address, username, token string, client *Cl
 func (sess *session) Close() error {
 	sess.cancel()
 
-	sess.mux.Close()
+	var errs []error
+	for _, child := range sess.subs {
+		err := child.Close()
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	if !sess.isSub {
```

Comment:
ah! so this was the main issue?

Reply:
Yep.

For some reason, bind fails with `cross-host bind not allowed` on CI. I had a guess that it complains about logins from 4 different hosts (running test session) with same login. However, when I tried to make username include current pid nothing changed, so looks like that's not that case.

Due to failing bind, mux was closed and it looked like "request hangs".

Reply:
ok.
I am not sure how much of this is "by design" and how much is "that's just the way it is".
perhaps something to clarify with upstream?